internal/plan/plan.go

```diff
@@ -178,6 +184,27 @@ func (s *Service) Merge(other *Service) {
 	if other.BackoffLimit.IsSet {
 		s.BackoffLimit = other.BackoffLimit
 	}
+	s.LogTargets = dedupStrings(s.LogTargets, other.LogTargets)
+}
+
+// dedupStrings merges arr1 and arr2 with deduplication (removing any duplicate
+// entries in the merged slice).
+func dedupStrings(arr1, arr2 []string) []string {
```

Comment:
Minor thing: I was originally going to suggest a different name for this, but then I realised this is doing two things: concatenating the arrays and then de-duping (but it's name only mentions the de-duping). I think it'd be simpler and easier to name to separate the two. Something like this:

```go
s.LogTargets = dedupStrings(append(s.LogTargets, other.LogTargets...))

...
// dedupStrings removes duplicates in slice, maintaining the order.
func dedupStrings(slice []string) []string { ... }
```